pkg/handler/context.go

```diff
@@ -19,8 +19,7 @@ type httpContext struct {
 
 func newContext(w http.ResponseWriter, r *http.Request) *httpContext {
 	return &httpContext{
-		// TODO: Try to reuse the request's context in the future
-		Context: context.Background(),
+		Context: r.Context(),
```

Comment:
This is not possible so easily as it breaks the S3Store (and possibly also gcsstore etc). The problem is that these stores run code using this context, which should also run even if the HTTP request has been aborted and thus the request context has been cancelled. The code in question is responsible for storing all of the data up to the moment that the request got stopped. And that is not possible if the context has been cancelled.

We haven't found a way so far to work around this problem but that is why the request context is not used for now. Let us know if you have any idea on how to solve this.